### PR TITLE
OIN updates: SCIM and Submit Guides

### PR DESCRIPTION
**Pull request by IanHakes-okta**
## Description:
- **What's changed?** 
Significant re-write of the Build a Provisioning Integration Guide for the OIN, and a new Submit your App guide for OIN manager/ISV portal. Existing content has been updated and restructured for flow and accuracy. UI descriptions and screen caps have been updated to reflect the current Okta Admin Console. Test buckets have been revamped with additional instructions and details, and all confirmed to work on Runscope. App analyst QA test file has been updated. 
- **Is this PR related to a Monolith release?**
2019.12.0 - contained all the UI changes and ISV portal updates that are described.

### Resolves:
* [OKTA-264202](https://oktainc.atlassian.net/browse/OKTA-264202)
* [OKTA-264937](https://oktainc.atlassian.net/browse/OKTA-264937)
* [OKTA-263401](https://oktainc.atlassian.net/browse/OKTA-263401)
* [OKTA-258236](https://oktainc.atlassian.net/browse/OKTA-258236)
* [OKTA-247001](https://oktainc.atlassian.net/browse/OKTA-247001)
* [OKTA-211008](https://oktainc.atlassian.net/browse/OKTA-211008)
* [OKTA-211007](https://oktainc.atlassian.net/browse/OKTA-211007)
* [OKTA-211000](https://oktainc.atlassian.net/browse/OKTA-211000)
* [OKTA-211003](https://oktainc.atlassian.net/browse/OKTA-211003)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+The SCIM spec specifies valid data formats for a given user profile attribute, however Okta does not rigorously validate that the customer has inputted values meeting those requirements to preserve flexibility.
+
+Therefore, data validation should be handled by your app's SCIM Server. In other words, when Okta provisions user profile to your app, it should check that the data is valid per their special requirements. Error messages sent in the response from your app will be surfaced to the Okta administrator via alerts and tasks in the Okta interface. You should also specify your data requirements in your config guide.
+
+**How much filtering support is needed?**
+
+The filtering capabilities in the SCIM protocol are pretty broad but the common filtering use case with Okta is quite narrow -- determine if a newly created Okta user already exists in your app based on a matching identifier . This means the eq (equals) operator is all you really need to support for now. We "might" eventually support other operators but don't right now.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: "narrow: determine" 

"identifier."

"the `eq` (equals) operator"

"all you ~really~ need to support ~for now~. ~We "might" eventually support other operators but don't right now.~"

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+Note that Okta only supports filtering via the eq operator on the SCIM userName attribute on the SCIM Server side. However, Okta can use any AppUser attribute on the Okta side for passing into the eq operator. Typically this would also be `appuser.userName`, but `appuser.email` or `appuser.randomAttribute` can also work.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: "`eq`" "`userName`" "`AppUser`"

File: packages/@okta/vuepress-site/docs/guides/submit-app/overview/index.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+title: Publishing an app integration with Okta Integration Network
+---
+
+This guide outlines the process by which software vendors (ISVs), Okta customers, and IT system-integrators (SI) can submit an app for review by Okta with the goal of publishing it to the Okta Integration Network (OIN).
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/submit-app/overview/index.md]: "independent software vendors"

Is this "SI" acronym actually used anywhere?

File: packages/@okta/vuepress-site/docs/guides/submit-app/prepare-for-submission/index.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Your submission must provide Okta with all the metadata needed to create a customized app for publication in the [Okta Integration Network](https://www.okta.com/integrations/).
+
+Okta analysts review the submission, create the customized app, run it through Okta internal quality assurance testing, and then publish it to the OIN.
+
+Complete the following steps before you submit your SCIM app:
+
+1. Check the [profile attributes and their mappings](/docs/guides/a-build-provisioning-integration/attribute-mapping/) for your app.
+1. [Test your Okta integration](/docs/guides/a-build-provisioning-integration/test-scim-app/) using the Okta SCIM CRUD test suite on Runscope.
+1. Prepare your [customer-facing configuration guide](/docs/guides/a-build-provisioning-integration/prepare-guide/).
+1. Optionally, [create a demo video](/docs/guides/a-build-provisioning-integration/prepare-guide/#create-a-demo-video) showing a working integration.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/submit-app/prepare-for-submission/index.md]: I'd argue that the link doesn't really provide any additional information, and so isn't real worth linking to.

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+  1. Click **OK** to confirm that you want to remove the mapping for the attribute you selected.
+    ![Remove Mapping](/img/oin/scim_check-attributes-3.png "Remove Mapping")
+
+  1. After removing the mapping for the unwanted attributes, click **To Okta** under the settings section.
+    ![Provisioning to Okta tab](/img/oin/scim_check-attributes-4.png "Provisioning Tab: Provisioning to Okta")
+
+  Repeat Step 3 and 4 until you remove all the mappings for the attributes you want to delete.
+
+B. Deleting attributes from your attribute list
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md]: Fixed.

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+  1. In the Profile Editor, scroll down to the attribute list.
+  
+  1. Look for the attribute that you want to delete, and click **X**.
+    ![Profile Editor - Remove Attribute](/img/oin/scim_check-attributes-7.png "Profile Editor - Remove Attribute")
+
+  1. Click **Delete Attribute** to confirm that you want to remove the attribute.
+    ![Profile Editor - Delete Attribute](/img/oin/scim_check-attributes-8.png "Profile Editor - Delete Attribute")
+
+## Adding Attributes
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md]: Fixed here and in "Mapping" heading title.

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+1. Scroll to the **Attribute Mappings** section. Click **Go to Profile Editor**.
+  ![Attribute Mappings - Profile Editor](/img/oin/scim_check-attributes-6.png "Attribute Mappings - Profile Editor")
+
+1. In the Profile Editor, click **Add Attribute**.
+  ![Profile Editor - Add Attribute](/img/oin/scim_check-attributes-10.png "Profile Editor - Add Attribute")
+  Enter the information for the new attribute that you’re adding and then click Save. For example:
+  ![Profile Editor - Add Attribute Dialog](/img/oin/scim_check-attributes-11.png "Profile Editor - Add Attribute Dialog")
+
+   >**Note:** The **Scope** property determines whether the attribute you are adding can be assigned at a Group level or just per user. If you would like your admins to be able to assign a value for this attribute at a Group level, do NOT check the **User personal** checkbox.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md]: I am okay if we leave it vanilla.
Changed "not" to lower case. 
Also changed "Group" to "group" as we are discussing it in general terms, not specifically referencing an object.

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+1. From the Admin Console, open your SCIM app.
+
+1. Go to the **Provisioning** tab. Under the **SETTINGS** section, click **To App**.
+  ![Provisioning to App tab](/img/oin/scim_check-attributes-1.png "Provisioning Tab: Provisioning to App")
+
+1. Scroll to the **Attribute Mappings** section. Look for the attribute that you want to update and click **Edit**.
+  ![Attribute Mappings - Edit Attribute](/img/oin/scim_check-attributes-13.png "Attribute Mappings - Edit Attribute")
+
+1. In the dialog that appears, there are two drop-down fields. On the first drop-down, select **Map from Okta Profile**. On the second drop-down, choose the Okta profile attribute that you would like to map the SCIM attribute from. Click Save.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md]: Fixed

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+## Attribute support
+
+You only want to include the attributes that you support in your current user schema. To ensure that the attributes are being sent properly to and from Okta:
+
+1. When assigning a user to the SCIM app you added in your dev org, ensure that all expected attributes are populated for that user.
+
+1. After the user is pushed to your SCIM app, check that all attributes are populated on your end.
+
+1. If your app supports **User Imports**, try importing one user from your app. Check the imported user and ensure that the values for supported attributes are reflected in that imported user's account in Okta.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/attribute-mapping/index.md]: Change made.

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+- [SCIM Reference](/docs/reference/scim/)
+
+If you haven't yet implemented SCIM, Okta recommends that you use Version 2.0 of the SCIM protocol. Also, an important part of the planning process is determining which Okta provisioning features your SCIM API can or should support.
+
+At a minimum, Okta requires that your SCIM 2.0 API implements the following features.
+
+### API Endpoints
+
+The API endpoint for your SCIM API **must** be secured using the [Transport Layer Security](https://tools.ietf.org/html/rfc5246) protocol. Connections through this secure layer are routed by using the `https://` prefix for your URL.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md]: Fixed.

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+- `https://system-admin.okta.com/admin/app/cpc/{appName}/oauth/callback`
+- `https://system-admin.okta-emea.com/admin/app/cpc/{appName}/oauth/callback`
+- `https://system-admin.oktapreview.com/admin/app/cpc/{appName}/oauth/callback`
+- `https://system-admin.trexcloud.com/admin/app/cpc/{appName}/oauth/callback`
+- `http://system-admin.okta1.com:1802/admin/app/cpc/{appName}/oauth/callback`
+
+where the `{appName}` is an identifier provided to you after your integration is submitted and processed by Okta.
+
+Okta doesn't support OAuth 2.0 [Resource Owner Password Credentials grant flows](https://tools.ietf.org/html/rfc6749#section-1.3.3) for securing a SCIM API.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md]: Removed for clarity

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+- User ID: `userName`
+- First Name: `name.givenName`
+- Last Name: `name.familyName`
+- Email: `emails`
+
+>**Note:** Keep your User ID value distinct from the user's email address. Many systems use an email address as a user identifier, but this is not recommended, as email addresses often change. Using a distinct User ID to identify user resources prevents future issues.
+
+Note that Okta supports more than those four user attributes. However, these are the base attributes that you must support.  The full schema of user attributes supported by SCIM 2.0 is described in [section 4 of RFC 7643](https://tools.ietf.org/html/rfc7643#section-4).
+
+If your integration supports user attributes beyond those four base attributes, you can add support for additional attributes to your SCIM API. In some cases, you might need to configure Okta to map non-standard user attributes into the user profile for your application. See the **Add custom attributes to an Okta user profile** section in the [Work with Okta user profiles and attributes](https://help.okta.com/en/prod/okta_help_CSH.htm#ext_Directory_Profile_Editor_Tasks) topic in the Okta product documentation.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md]: Replaced bold marks with quotes:

> See the "Add custom attributes to an Okta user profile" section...

File: packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+Okta supports lifecycle provisioning through version 2.0 and version 1.1 of the SCIM protocol.
+
+If your service already supports the SCIM protocol, it is important that you review the Okta SCIM reference documentation to understand the specifics of how Okta implements the SCIM protocol:
+
+- [SCIM Reference](/docs/reference/scim/)
+
+If you haven't yet implemented SCIM, Okta recommends that you use Version 2.0 of the SCIM protocol. Also, an important part of the planning process is determining which Okta provisioning features your SCIM API can or should support.
+
+At a minimum, Okta requires that your SCIM 2.0 API implements the following features.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-provisioning-integration/prepare-api/index.md]: Good idea. Done.

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+Okta has seen significant SCIM momentum in the market amongst our network of app developers over the past year. Hot apps like [Slack](https://api.slack.com/scim) and [Lucidchart](https://www.lucidchart.com/techblog/2016/08/04/an-implementers-overview-managing-cloud-identity-with-scim/) are supporting SCIM as well established software companies like [Huddle](https://github.com/Huddle/huddle-apis/wiki/Integrating%20with%20SCIM) and [Citrix](https://developer.citrixonline.com/implementing-scim-apis).
+
+Okta is invested in SCIM provisioning and launched our own SCIM provisioning developer program. The SCIM standard is strong and is run by Salesforce, Google, and Sailpoint (Okta is also a contributor).
+
+**How should I be managing authentication to my SCIM API?**
+
+Okta recommends using the OAuth 2.0 Authorization Code Grant Flow (aka "three-legged OAuth"). Okta doesn't support the Client Credentials or Resource Owner Password Credentials Authorization grant flows. The Authorization Code Grant Flow is more common in SaaS/cloud and is also more secure. In addition to OAuth, Okta also supports basic auth and header token auth options.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: Removed

File: packages/@okta/vuepress-site/docs/guides/submit-app/overview/index.md (hunk 3 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/overview/index.md]: Pulled

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+If you have any technical questions, you can submit them to <developers@okta.com> or post your questions on our [Developer Forum](https://devforum.okta.com/).
+
+
+**By following the guidance here, am I building an integration that only works with Okta? What about other identity vendors?**
+
+Absolutely not. Our goal is to help you identity-enable your application using industry standards. The guidance offered here for SAML and SCIM allows you to integrate with customers using other identity solutions.
+
+**My customer is asking for AD (Active Directory) integration. If I integrate with Okta, can I connect to my customer's on-premise directory?**
+
+Yes. This is one of the key benefits of developing a pre-built integration with Okta - you can leverage our existing integrations with directories so you don't have to. By integrating with Okta (for single sign-on and provisioning), you effectively have the ability to integrate with your customer's on-premise AD or LDAP infrastructure for authentication (log into your cloud app with their corporate password), authorization (use details like AD groups to drive access rights), and provisioning policies.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: Changed to "One of the key benefits of developing a pre-built integration with Okta is that you can leverage our existing active directory support."

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+
+**What is Secure Web Authentication (SWA)?**
+
+SWA was developed by Okta to provide single sign-on for apps that do not support federated sign-on methods. Users can still sign in directly through the application and then enter their credentials for these apps on their Okta homepage. These credentials are stored such that users can access their apps with a single sign-on. When users first sign-in to a SWA app from their homepage, they see a pop-up message asking if they were able to sign-in successfully.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: Changed the last two "sign-in"s to "sign on"

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+Have you submitted an app but have not seen a change in review status in the OIN Manager? Email <developers@okta.com>.
+
+**I'm an ISV setting up a SAML 2.0 app using the App Wizard and we have different domains for each customer. How do you manage these types of situations?**
+
+Currently, the App Wizard does not support custom domains, but public facing integrations can still have this functionality. Create an app integration as you normally would using the App Wizard. Once you get to the step of submitting your app integration via the [OIN Manager](https://oinmanager.okta.com), you will be asked to fill out a form (see image below); be sure to select “No” and provide all relevant information. We will add the ability for customers to set a custom domain in the customer facing integration.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: Changed to "Currently, the App Wizard does not support custom domains, but public-facing integrations can still have this functionality. Create an app integration as you normally would using the App Wizard. After you get to the step of submitting your app integration through the [OIN Manager](https://oinmanager.okta.com), you will be asked to fill out a form with all the relevant information." 
I will have Eduardo review this question when he's back from PTO.

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+
+Currently, the App Wizard does not support extra login fields. Create an app using the Plug-in (SWA) Template Application. In order to promote your Template App to the Okta Integration Network, please email a screenshot of the configured app details to <developers@okta.com> with your app name in the subject line.
+
+**Does Okta support single logout or single sign-out (SAML protocol)?**
+
+Yes. For more information, see the Advanced Settings section of the [Using the App Integration Wizard: SAML App Wizard](https://help.okta.com/en/prod/okta_help_CSH.htm#ext_Apps_App_Integration_Wizard).
+
+**Is the IDP session time out a setting that an Okta administrator can change? And if so, can it be changed on a per application basis, or is it a global setting for all of the user's applications?**
+
+Yes, the session time out default is two hours but the Okta administrator can customized the default by hour or minute. This session time out is an IDP setting and therefore, it is global and applies to all applications in an Okta org.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: Added a link to the Security Policies documentation.

File: packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+
+Okta has seen significant SCIM momentum in the market amongst our network of app developers over the past year. Hot apps like [Slack](https://api.slack.com/scim) and [Lucidchart](https://www.lucidchart.com/techblog/2016/08/04/an-implementers-overview-managing-cloud-identity-with-scim/) are supporting SCIM as well established software companies like [Huddle](https://github.com/Huddle/huddle-apis/wiki/Integrating%20with%20SCIM) and [Citrix](https://developer.citrixonline.com/implementing-scim-apis).
+
+Okta is invested in SCIM provisioning and launched our own SCIM provisioning developer program. The SCIM standard is strong and is run by Salesforce, Google, and Sailpoint (Okta is also a contributor).
+
+**How should I be managing authentication to my SCIM API?**
+
+Okta recommends using the OAuth 2.0 Authorization Code grant flow. Okta doesn't support the Client Credentials or Resource Owner Password Credentials Authorization grant flows. The Authorization Code grant flow is more common in SaaS/cloud and is also more secure. In addition to OAuth, Okta also supports basic auth and header token auth options.
```
[the remark this answers is not in the document]
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/faqs/index.md]: Added in a link to /docs/guides/implement-auth-code/ for the OAuth 2.0 auth code grant flow entry (as this is primarily how we want to direct ISVs to connect their SICM APIs

File: packages/@okta/vuepress-site/docs/guides/submit-app/prepare-for-submission/index.md (hunk 4 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/submit-app/prepare-for-submission/index.md]: Link removed.

**Pull request merged**